src/kw_config_loader.sh

```diff
@@ -27,14 +28,41 @@ function show_variables()
     has_local_config_path="No"
   fi
 
+  if [[ "$1" == 'TEST_MODE' ]]; then
```

Comment:
I did not understand the value of TEST_MODE in this function.

Reply:
Test mode assures all possible configurations are printed, regardless if they are defined or not.

Reply:
Usually, I use the test mode for validating command sequences or other things that might be hard to test. Here, I'm struggling to see the value of having this TEST_MODE. For example, take a look at some of the tests associated with mk.

Reply:
Our first approach for this test was to define a custom `configurations` array, including every possible valid configuration option for kw. This way we can test if all the valid options are being printed by `show_variables`, by grepping it's output. Now we check that the set of configuration options that `show_variables` is capable of printing is not smaller than the set of valid configuration options.

Consider the case where a configuration option is removed from kw, and is no longer valid. Consider that the developer who removed that option forgot to remove it from `show_variables`. Now if a user defines this obsolete option manually in his `kworkflow.config`, `kw vars` will still print it as a valid option! We want to avoid that and we want our test to cover this case. (keep in mind that if an option is not defined, it will not be printed by `show_variables`). But how can we check if some undefined option could potentially be printed by `show_variables`? That is, how can we guarantee that the set of configuration option that `show_variables` is capable of printing is not bigger, i.e., does not contain more elements than the set of valid configuration options for kw? We would have to test every possible option, but that is not feasible. This is the purpose of TEST_MODE: it alters the default behavior of not printing options if they are undefined: in TEST_MODE, `show_variables` will print every configuration option that it is capable of printing. That way we can test this edge case.